servo/src/server/routes/mappings.rs

```diff
@@ -18,6 +16,261 @@ pub struct GetMappingResponse {
   pub mappings: Vec<NodeMapping>,
 }
 
+const PYTHON_KEYWORDS : [&str; 70] = [
+  "abs", "aiter", "all", "anext", "any", "ascii", "bin", "bool", "breakpoint", 
+  "bytearray", "bytes", "callable", "chr", "classmethod", "compile", "complex", 
+  "delattr", "dict", "dir", "divmod", "enumerate", "eval", "exec", "filter", 
+  "float", "format", "frozenset", "getattr", "globals", "hasattr", "hash", 
+  "help", "hex", "id", "input", "int", "isinstance", "issubclass", "iter", 
+  "len", "list", "locals", "map", "max", "memoryview", "min", "next", "object", 
+  "oct", "open", "ord", "pow", "print", "property", "range", "repr", "reversed",
+  "round", "set", "setattr", "slice", "sorted", "staticmethod", "str", "sum", 
+  "super", "tuple", "type", "vars", "zip"
+];
+    
+/// Returns where an identifier is a preexisting python keyword
+/// (Only checks base python functions)
+fn is_python_keyword(identifier : &String) -> bool {
+  PYTHON_KEYWORDS.contains(&identifier.as_str())
+}
+
+/// Validates the text_id of a mapping against python variable naming 
+/// conventions
+fn validate_mapping_identifier(mapping : &NodeMapping)-> ServerResult<()> {
+  let text_id : &String = &mapping.text_id;
+
+  // Mapping's text_id should not be a python keyword
+  if is_python_keyword(text_id) {
+    return Err(bad_request(
+      format!("Mapping name \"{}\" is already a python keyword", text_id)
+    ));
+  }
+
+  // Mapping's text_id should not be empty
+  if text_id.is_empty() {
+    return Err(bad_request(
+      // As there is no name, identify the mapping through it's parameters
+      // May change to sensor_type + board_id + channel if needed
+      format!("A mapping had an empty name field!\n{:#?}", mapping)
+    ));
+  }
+
+  
+  // all characters must be alphanumber or '_'
+  for character in text_id.chars() {
+    // While this is checked covered in future code,
+    // people testing constantly mess this up, so there is a specific
+    // Error message when using '-' in mapping identifiers
+    if character == '-' {
+      return Err(bad_request(
+        format!("mapping name \"{}\" {}", 
+          text_id,
+          "contains the symbol '-', which is NOT ALLOWED AS IT BREAKS SEQUENCES"
+        )
+      ));
+    }
+    
+    if character == ' ' {
+      return Err(bad_request(
+        format!("mapping name \"{}\" {}", 
+          text_id,
+          "contains a space, which is NOT ALLOWED"
+        )
+      ));
+    }
+    // Actually check if alphanumeric or '_'
+    if !(character.is_alphanumeric() || character == '_') {
+      return Err(bad_request(
+        format!("mapping name \"{}\" contains invalid character '{}'", 
+          text_id, character,)
+      ));
+    }
+  }
+
+  // First character cannot be a digit
+  let first_character : char = text_id.chars().next()
```

Comment:
Superfluous type annotation.

Reply:
Also, I'd get the first character at the same time as checking if the string is empty. Get the first character, and if that fails, then return the error that the string is empty.